src/period-calculation/period-types.ts

```diff
@@ -0,0 +1,90 @@
+import { PeriodIdentifier } from './types'
+
+export const FIXED_PERIOD_TYPES: { [name: string]: PeriodIdentifier } = {
+    DAILY: 'Daily',
+    WEEKLY: 'Weekly',
+    WEEKLYWED: 'WeeklyWednesday',
+    WEEKLYTHU: 'WeeklyThursday',
+    WEEKLYSAT: 'WeeklySaturday',
+    WEEKLYSUN: 'WeeklySunday',
+    BIWEEKLY: 'BiWeekly',
+    MONTHLY: 'Monthly',
+    BIMONTHLY: 'BiMonthly',
+    QUARTERLY: 'Quarterly',
+    QUARTERLYNOV: 'QuarterlyNov', // used in Ethiopi: 'SixMonthly',
+    SIXMONTHLY: 'SixMonthly',
+    SIXMONTHLYAPR: 'SixMonthlyApril',
+    SIXMONTHLYNOV: 'SixMonthlyNov', // used in Ethiopia
+    YEARLY: 'Yearly',
+    FYAPR: 'FinancialApril',
+    FYJUL: 'FinancialJuly',
+    FYOCT: 'FinancialOct',
+    FYNOV: 'FinancialNov',
+}
+
+export const WEEKLY_FIXED_PERIOD_TYPES: PeriodIdentifier[] = [
+    FIXED_PERIOD_TYPES.WEEKLY,
+    FIXED_PERIOD_TYPES.WEEKLYWED,
+    FIXED_PERIOD_TYPES.WEEKLYTHU,
+    FIXED_PERIOD_TYPES.WEEKLYSAT,
+    FIXED_PERIOD_TYPES.WEEKLYSUN,
+    FIXED_PERIOD_TYPES.BIWEEKLY,
+]
+
+export const MONTLY_FIXED_PERIOD_TYPES: PeriodIdentifier[] = [
+    FIXED_PERIOD_TYPES.MONTHLY,
+    FIXED_PERIOD_TYPES.BIMONTHLY,
+    FIXED_PERIOD_TYPES.QUARTERLY,
+    FIXED_PERIOD_TYPES.QUARTERLYNOV,
+    FIXED_PERIOD_TYPES.SIXMONTHLY,
+    FIXED_PERIOD_TYPES.SIXMONTHLYAPR,
+    FIXED_PERIOD_TYPES.SIXMONTHLYNOV,
+]
+
+export const QUARTERLY_FIXED_PERIOD_TYPES: PeriodIdentifier[] = [
+    FIXED_PERIOD_TYPES.QUARTERLY,
+    FIXED_PERIOD_TYPES.QUARTERLYNOV,
+]
+
+export const SIXMONTHLY_FIXED_PERIOD_TYPES: PeriodIdentifier[] = [
+    FIXED_PERIOD_TYPES.SIXMONTHLY,
+    FIXED_PERIOD_TYPES.SIXMONTHLYAPR,
+    FIXED_PERIOD_TYPES.SIXMONTHLYNOV,
+]
+
+export const MONTHLY_STANDARD_FIXED_PERIOD_TYPES: PeriodIdentifier[] = [
+    FIXED_PERIOD_TYPES.MONTHLY,
+    FIXED_PERIOD_TYPES.BIMONTHLY,
+    FIXED_PERIOD_TYPES.QUARTERLY,
+    FIXED_PERIOD_TYPES.SIXMONTHLY,
+]
+
+export const MONTHLY_OFFSET_FIXED_PERIOD_TYPES: PeriodIdentifier[] = [
+    FIXED_PERIOD_TYPES.QUARTERLYNOV,
+    FIXED_PERIOD_TYPES.SIXMONTHLYAPR,
+    FIXED_PERIOD_TYPES.SIXMONTHLYNOV,
+]
+
+export const MULTI_MONTH_FIXED_PERIOD_TYPES: PeriodIdentifier[] = [
+    FIXED_PERIOD_TYPES.BIMONTHLY,
+    FIXED_PERIOD_TYPES.QUARTERLY,
+    FIXED_PERIOD_TYPES.QUARTERLYNOV,
+    FIXED_PERIOD_TYPES.SIXMONTHLY,
+    FIXED_PERIOD_TYPES.SIXMONTHLYAPR,
+    FIXED_PERIOD_TYPES.SIXMONTHLYNOV,
+]
+
+export const FINANCIAL_YEAR_FIXED_PERIOD_TYPES: PeriodIdentifier[] = [
+    FIXED_PERIOD_TYPES.FYAPR,
+    FIXED_PERIOD_TYPES.FYJUL,
+    FIXED_PERIOD_TYPES.FYOCT,
+    FIXED_PERIOD_TYPES.FYNOV,
+]
+
+export const YEARLY_FIXED_PERIOD_TYPES: PeriodIdentifier[] = [
+    FIXED_PERIOD_TYPES.YEARLY,
+    FIXED_PERIOD_TYPES.FYAPR,
+    FIXED_PERIOD_TYPES.FYJUL,
+    FIXED_PERIOD_TYPES.FYOCT,
+    FIXED_PERIOD_TYPES.FYNOV,
+]
```

Comment:
These can just be arrays of strings, since. The type declaration will prevent us from using typos.

Reply:
This completely changed, see [my comment further above](https://github.com/dhis2/multi-calendar-dates/pull/12#discussion_r1107934306).